### PR TITLE
[ML] Add PyTorch setup instructions for Linux

### PR DESCRIPTION
**Pull request by davidkyle**
Instructions for installing the dependencies and building PyTorch on Linux

Follows the same commands used to build the docker image in #1678

### CODE REVIEW

File: build-setup/linux.md [hunk 1]
```diff
@@ -237,6 +237,67 @@ sudo make install
...
 to complete the build.
 
+### CMake
+
+CMake version 3.5 is the minimum required to build PyTorch. Download version 3.19.3 from <https://github.com/Kitware/CMake/releases/download/v3.19.3/cmake-3.19.3-Linux-x86_64.sh> and install:
+
+```
+chmod +x cmake-3.19.3-Linux-x86_64.sh
+mkdir /usr/local/cmake
+./cmake-3.19.3-Linux-x86_64.sh --skip-license --prefix=/usr/local/cmake
```
Comment (droberts195) [inline build-setup/linux.md:247]:
```suggestion
sudo mkdir /usr/local/cmake
sudo ./cmake-3.19.3-Linux-x86_64.sh --skip-license --prefix=/usr/local/cmake
```

The Dockerfile runs as root, whereas the average user with a Linux laptop won't be, so best to include the `sudo` in these instructions.

File: build-setup/linux.md [hunk 2]
```diff
@@ -237,6 +237,67 @@ sudo make install
...
+```  
+
+In your `.bashrc` file add CMake to `PATH`:
+
+```
+PATH=$PATH:/usr/local/cmake/bin
+```
```
Comment (droberts195) [inline build-setup/linux.md:254]: You could include this bit on line 20 so the person working through the instructions can do that all in one go.

File: build-setup/linux.md [hunk 3]
```diff
@@ -237,6 +237,67 @@ sudo make install
...
+
+### Python 
+
+PyTorch currently requires Python 3.6, 3.7 or 3.8, and version 3.7 appears to cause fewest problems in their test status matrix, so we use that. If your system does not have a requisite version of Python install it with a package manager or build the last 3.7 release from source by downloading `Python-3.7.9.tgz` from <https://www.python.org/ftp/python/3.7.9/Python-3.7.9.tgz> then extract and build:
+
+```
+tar -xzf Python-3.7.9.tgz
+cd Python-3.7.9/
+./configure --enable-optimizations
+make altinstall
```
Comment (droberts195) [inline build-setup/linux.md:264]: Does this install to a directory that requires root access?  If so it needs `sudo`, but for all the other packages we build from source we do the bulk of the build as non-root and just the install as root, i.e. this sequence occurs multiple times in this file:

```
make
sudo make install
```

Maybe there's an equivalent for Python like:

```
make
sudo make altinstall
```

Or if not then probably the whole build needs to be under `sudo`.

File: build-setup/linux.md [hunk 4]
```diff
@@ -237,6 +237,67 @@ sudo make install
...
+/usr/local/bin/python3.7 setup.py install
+```
+
+Once built copy headers and libraries to system directories:
+
+```
+mkdir /usr/local/gcc93/include/pytorch
+cp -r torch/include/* /usr/local/gcc93/include/pytorch/ 
+cp torch/lib/libtorch_cpu.so /usr/local/gcc93/lib 
+cp torch/lib/libc10.so /usr/local/gcc93/lib
```
Comment (droberts195) [inline build-setup/linux.md:298]:
```suggestion
sudo mkdir /usr/local/gcc93/include/pytorch
sudo cp -r torch/include/* /usr/local/gcc93/include/pytorch/ 
sudo cp torch/lib/libtorch_cpu.so /usr/local/gcc93/lib 
sudo cp torch/lib/libc10.so /usr/local/gcc93/lib 
```

File: build-setup/linux.md [hunk 5]
```diff
@@ -237,6 +237,68 @@ sudo make install
...
+### Python 
+
+PyTorch currently requires Python 3.6, 3.7 or 3.8, and version 3.7 appears to cause fewest problems in their test status matrix, so we use that. If your system does not have a requisite version of Python install it with a package manager or build the last 3.7 release from source by downloading `Python-3.7.9.tgz` from <https://www.python.org/ftp/python/3.7.9/Python-3.7.9.tgz> then extract and build:
+
+```
+tar -xzf Python-3.7.9.tgz
+cd Python-3.7.9/
+./configure --enable-optimizations
+make
+sudo make altinstall
```
Comment (davidkyle) [inline build-setup/linux.md:265]: Reading the make file the `altinstall` target does not overwrite any current installations by updating softlinks etc. We will use altinstall and be explicit when invoking python

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**